src/main.rs

```diff
@@ -295,7 +295,8 @@ fn run() -> Result<ExitCode> {
         max_results: matches
             .value_of("max-results")
             .and_then(|n| usize::from_str_radix(n, 10).ok())
-            .filter(|&n| n != 0),
+            .filter(|&n| n != 0)
+            .or_else(|| if matches.is_present("max-one-result") { Some(1) } else { None }),
```

Comment:
:+1: 

Just a side note: this type of pattern
``` rust
if boolean_expression { Some(value) } else { None }
```
can possibly be written as `boolean_expression.then_some(value)` in the future: https://doc.rust-lang.org/std/primitive.bool.html#method.then_some

This line would then simplify to:
``` rust
.or_else(|| if matches.is_present("max-one-result").then_some(1)), 
```

Reply:
Yes, I was looking for something like this and found the method you mentioned, just to realize that it is currently experimental. Too bad.